src/Map/assets/src/abstract_map_controller.ts

```diff
@@ -106,32 +122,65 @@ export default abstract class<
         return marker;
     }
 
+    createPolygon(definition: PolygonDefinition<PolygonOptions, InfoWindowOptions>): Polygon {
+        this.dispatchEvent('polygon:before-create', { definition });
+        const polygon = this.doCreatePolygon(definition);
+        this.dispatchEvent('polygon:after-create', { polygon });
+        this.polygons.push(polygon);
+        return polygon;
+    }
+
     protected abstract doCreateMarker(definition: MarkerDefinition<MarkerOptions, InfoWindowOptions>): Marker;
+    protected abstract doCreatePolygon(definition: PolygonDefinition<PolygonOptions, InfoWindowOptions>): Polygon;
 
-    protected createInfoWindow({
+    protected createInfoWindowMarker({
         definition,
         marker,
     }: {
         definition: MarkerDefinition<MarkerOptions, InfoWindowOptions>['infoWindow'];
         marker: Marker;
     }): InfoWindow {
-        this.dispatchEvent('info-window:before-create', { definition, marker });
-        const infoWindow = this.doCreateInfoWindow({ definition, marker });
-        this.dispatchEvent('info-window:after-create', { infoWindow, marker });
+        this.dispatchEvent('info-window-marker:before-create', { definition, marker });
+        const infoWindow = this.doCreateInfoWindowMarker({ definition, marker });
+        this.dispatchEvent('info-window-marker:after-create', { infoWindow, marker });
 
         this.infoWindows.push(infoWindow);
 
         return infoWindow;
     }
 
-    protected abstract doCreateInfoWindow({
+    protected createInfoWindowPolygon({
+        definition,
+        polygon,
+    }: {
+        definition: PolygonDefinition<PolygonOptions, InfoWindowOptions>['infoWindow'];
+        polygon: Polygon;
+    }): InfoWindow {
+        this.dispatchEvent('info-window-polygon:before-create', { definition, polygon });
+        const infoWindow = this.doCreateInfoWindowPolygon({ definition, polygon });
+        this.dispatchEvent('info-window-polygon:after-create', { infoWindow, polygon });
+
+        this.infoWindows.push(infoWindow);
+
+        return infoWindow;
+    }
```

Comment:
I don't think we want to introduce a new very similar method to create an info window **but** for a polygon.

Instead, can you keep the previous code but add a new `polygon` parameter?

This way we won't have to break apps using `info-window:before-create` / `info-window:after-create` events (even if the component is experimental).

Reply:
With TypeScript, you can achieve that by using functions overloading:
- use a function definition for `marker` and `definition: MarkerDefinition<MarkerOptions, InfoWindowOptions>['infoWindow']`
- use a function definition for `polygon` and `definition: PolygonDefinition<PolygonOptions, InfoWindowOptions>['infoWindow']`
- use a final definition mixing both previous function definitions